### PR TITLE
Remove deprecated `Entity` and rename `Persisted` types

### PR DESCRIPTION
**Pull request by thehabbos007**
## 🌟 What is the purpose of this PR?

This PR removes old `Entity` GraphQL types, the `Persisted` prefix for our custom types along with moving deprecated-for-sure typedefs to a "deprecated.typdef.ts"

Some changes to the frontend may break, but as far as I understand the code paths that have been removed weren't used since `dev/graph` anyway.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203301853897838/1203214689883093/f) _(internal)_

## 🔍 What does this change?

- See commit

### CODE REVIEW

File: packages/hash/api/codegen.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ generates:
         DataTypeWithMetadata: "@hashintel/hash-subgraph#DataTypeWithMetadata"
         PropertyTypeWithMetadata: "@hashintel/hash-subgraph#PropertyTypeWithMetadata"
         EntityTypeWithMetadata: "@hashintel/hash-subgraph#EntityTypeWithMetadata"
-        EntityWithMetadata: "@hashintel/hash-subgraph#EntityWithMetadata"
+        Entity: "@hashintel/hash-subgraph#EntityWithMetadata"
```
Comment (thehabbos007) [inline packages/hash/api/codegen.yml:39]: `@hashintel/hash-subgraph#EntityWithMetadata` is kept the same to prevent GQL name clashes

Comment (thehabbos007) [reply packages/hash/api/codegen.yml]: I didn't consider we define Entity as a scalar now. Complete rename in 5146d27288ef5281285dc6576ff12cad69c8d555

**Review by Alfred-Mountfield — APPROVED** [review]
Thanks Ahmad, I absolutely love these changes. I've ran the app locally and it all seems to work, I did my best to tackle the diff but renamings are always going to be hard due to the sheer scale.

Let's get this in 🎉

File: packages/hash/frontend/src/blocks/page/BlockContextMenu/LoadEntityMenuContent.tsx [hunk 2]
```diff
@@ -80,49 +71,8 @@ export const LoadEntityMenuContent: FunctionComponent<
     [blockView, blockEntityId],
   );
 
-  const filteredEntities = useMemo(() => {
-    const uniqueEntityIds = new Set();
-    return entities.filter((entity) => {
```
Comment (thehabbos007) [inline packages/hash/frontend/src/blocks/page/BlockContextMenu/LoadEntityMenuContent.tsx:85]: @CiaranMn  I want to point out we removed a fair bit of the top-level code for the `LoadEntityMenu` which I assume the data mapping demo was based on

Comment (CiaranMn) [reply packages/hash/frontend/src/blocks/page/BlockContextMenu/LoadEntityMenuContent.tsx:85]: It was the entities for the 'swap entity' context menu item (i.e. change which entity is loaded into a block) – I've captured a [task](https://app.asana.com/0/1201095311341924/1203438226339015/f) (internal) to fix it

Comment (thehabbos007) [reply packages/hash/frontend/src/blocks/page/BlockContextMenu/LoadEntityMenuContent.tsx:85]: Thank you :)

**Pull request merged**